### PR TITLE
fix(stringlist): allocate string before assignment

### PR DESCRIPTION
**Pull request by mjreno**

### CODE REVIEW

File: src/Utilities/CharString.f90 [hunk 1]
```diff
@@ -100,4 +101,11 @@ subroutine write_unformatted(this, unit, iostat, iomsg)
     end if
   end subroutine write_unformatted
 
+  function strlen(this) result(length)
+    class(CharacterStringType), intent(in) :: this
+    integer :: length
+
```
Comment (langevin-usgs) [inline src/Utilities/CharString.f90:107]: should we say:

```
if (allocated(this%charstring)) then
  length = len(this%charstring))
else
  length = 0
end if
```

File: src/Utilities/StringList.f90 [hunk 2]
```diff
@@ -63,11 +63,13 @@ function GetStringFromList(list, indx) result(string)
     class(*), pointer :: obj
     type(CharacterStringType), pointer :: charcont
     !
-    string = ''
     obj => list%GetItem(indx)
     charcont => CastAsCharacterStringType(obj)
     if (associated(charcont)) then
+      allocate (character(len=charcont%strlen()) :: string)
       string = charcont
```
Comment (langevin-usgs) [inline src/Utilities/StringList.f90:70]: I think this should be:

```
string(:) = charcont
```

but please verify

**Review by langevin-usgs — CHANGES_REQUESTED** [review]
I was hoping we could overload the intrinsic len function for this, but I couldn't get it to work.  

I made a couple minor suggestions.

File: src/Utilities/CharString.f90 (hunk 1 above)
Comment (mjreno) [reply src/Utilities/CharString.f90:107]: I was thinking len should report 0 if not allocated but should have verified.  I'll make the change.

File: src/Utilities/StringList.f90 (hunk 2 above)
Comment (mjreno) [reply src/Utilities/StringList.f90:70]: Will do.

Comment (mjreno) [reply src/Utilities/StringList.f90:70]: The suggestion is to replace line 70 right?

**Review by langevin-usgs — APPROVED** [review]

**Pull request merged**